### PR TITLE
circulation: send emails on loan actions

### PR DESCRIPTION
**Pull request by equadon**
* Send emails when the loan status changes
* Configurable emails with `BlockTemplatedMessage` class
* Set sender, CC and BCC in `config.py`

### CODE REVIEW

File: invenio_app_ils/circulation/mail.py
Comment (ntarocco) [inline invenio_app_ils/circulation/mail.py]: I do not understand this `PENDING` - `LOAN_REQUESTED`, if pending it is requested :)

**Review by ntarocco — DISMISSED** [review]
Fantastic, great job!

**Comment by ntarocco** [comment]
@zzacharo @kprzerwa @topless I encourage you to review because we will have to customize templates and use this a lot!

File: invenio_app_ils/circulation/mail.py
Comment (kpsherva) [reply invenio_app_ils/circulation/mail.py]: I would leave `PENDING` for the name, because requested might be misleading, that there will be no waiting for approval

File: invenio_app_ils/circulation/mail.py
Comment (kpsherva) [inline invenio_app_ils/circulation/mail.py]: Do we manage somehow the case when the extension is blocked because the count exceeded max extensions allowed?

**Review by kpsherva — DISMISSED** [review]

**Comment by kpsherva** [comment]
needs rebase

File: invenio_app_ils/circulation/mail.py
Comment (equadon) [reply invenio_app_ils/circulation/mail.py]: Maybe use the name `PENDING` for pending requests that has `loanable` items and `PENDING_NO_ITEMS` for when there are 0 `loanable` items?

**Review by zzacharo — DISMISSED** [review]

**Review by ntarocco — DISMISSED** [review]

**Pull request merged**
